### ISSUE TITLE
Support official formula

### ISSUE THREAD

**Issue by uncenter** [oven-sh/homebrew-bun#8]
I'm glad that there is a formula but I'm surprised it is on this tap instead of https://github.com/Homebrew/homebrew-core. A project of Bun's popularity should totally have its own formula!

**Comment by Electroid** [comment]
Yes! This is something we wanted to do before 1.0, but will to do soon. The reason this hasn't happened is because to become an official formula, you can't provide a binary, homebrew must build Bun from scratch. In the past, Bun needed a lot of resources to build, because it is mostly written in Zig. This has changed and it should be easier now.

**Comment by vieira** [comment]
There is already some work under way in https://github.com/Homebrew/homebrew-core/pull/142016, if someone from bun can help push it forward.

**Comment by jylin** [comment]
FWIW I prefer the binary releases, so I hope this tap is kept even if bun is merged into core homebrew.
Their rule is too strict tbqh, and for things like node/bun, there isn't that much value of compiling from source.

**Comment by masterflitzer** [comment]
why? compiling from source proofs that the source is legit, why would there be no value for node/bun but for everything else?

**Comment by jylin** [comment]
They're more expensive to build and checksums should be enough. Do you build your web browser from source?
Many homebrew formulas have "bottles" anyway, so they aren't even typically built from source.

**Comment by masterflitzer** [comment]
I thought the bottles in homebrew/core are build from source automatically?

also not everyone has to build it from source, brew builds it once and we can install the binary, that's like a big point of even using brew and I think it's good to enforce it at least in the core repo

**Comment by botantony** [comment]
Does Bun still plan to support compilation from source without pre-built binaries?

**Comment by idleberg** [comment]
I don't think it has been mentioned before – one of the key advantages of an official formula is being able to depend on `bun` during build. Third-party taps are not supported.

**Example**

```rb
def install
  system "bun", "build", "--compile", "--outfile=${bin}/mybin", "src/main.ts"
end
```

**Comment by frou** [comment]
There does seem to be an official **homebrew-core** formula for Bun now:

- https://formulae.brew.sh/formula/bun#default
- https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/b/bun.rb
